lxqt-admin-time/timedatectl.cpp

```diff
@@ -0,0 +1,44 @@
+#include "timedatectl.h"
+#include <QProcess>
+#include <QDebug>
+
+TimeDateCtl::TimeDateCtl()
+{
+    QProcess timedatectl;
+    timedatectl.start(QStringLiteral("timedatectl"));
+    timedatectl.waitForFinished();
+    while(!timedatectl.atEnd()) {
+        QString line = timedatectl.readLine().trimmed();
+        int findTZ = line.indexOf(QLatin1String("Time zone:"));
+        if(findTZ != -1) {
+            findTZ += 10;
+            while(line[findTZ] == ' ')
+                ++findTZ;
+            int space = line.indexOf(' ', findTZ);
+            mTimeZone = line.mid(findTZ, space - findTZ);
+            break;
+        }
+    }
+    timedatectl.close();
```

Comment:
It might be better, more reliable, to directly use the dbus interface (`org.freedesktop.timedate1`) for this, after all that's pretty much all `timedatectl` is doing.